client/securedrop_client/logic.py

```diff
@@ -870,9 +874,12 @@ def on_message_download_failure(self, exception: DownloadException) -> None:
         try:
             message = storage.get_message(self.session, exception.uuid)
             self.message_download_failed.emit(message.source.uuid, message.uuid, str(message))
-        except Exception as e:
```

Comment:
Was it intentional to lose this generic exception handling?

Reply:
It was intentional - I think that the general exception handling here was specifically to guard against this edge case. I think we should have consistent logic in all of our callbacks (so either swallow all Exceptions in all the callbacks, or, still my preference (modulo the changes you requested), wrap _unforseen_ SQLAlchemy exceptions in our own class and try-catch those anywhere we perform db operations.  If it feels too risky I can revert the change.

Reply:
I stand by my decision- looking at the generic exception handling here, it was only handling `storage.get_message`, not the session commit or anything, so imo we handle all sqlalchemy errors (followup pr?) or we commit to the bit :)

Reply:
Makes sense and agreed!